politeiawww/convert.go

```diff
@@ -1824,3 +1826,94 @@ func filterDomainInvoice(inv *cms.InvoiceRecord) cms.InvoiceRecord {
 
 	return *inv
 }
+
+func convertCodeStatsFromDatabase(userCodeStats []user.CodeStats) []cms.CodeStats {
+	cmsCodeStats := make([]cms.CodeStats, 0, len(userCodeStats))
+	for _, codeStat := range userCodeStats {
+		prs := make([]string, 0, len(codeStat.PRs))
+		reviews := make([]string, 0, len(codeStat.Reviews))
+		for _, pr := range codeStat.PRs {
+			if pr == "" {
+				continue
+			}
+			prs = append(prs, pr)
+		}
+		for _, review := range codeStat.Reviews {
+			if review == "" {
+				continue
+			}
+			reviews = append(reviews, review)
+		}
+		cmsCodeStat := cms.CodeStats{
+			Month:           codeStat.Month,
+			Year:            codeStat.Year,
+			Repository:      codeStat.Repository,
+			PRs:             prs,
+			Reviews:         reviews,
+			MergeAdditions:  codeStat.MergedAdditions,
+			MergeDeletions:  codeStat.MergedDeletions,
+			ReviewAdditions: codeStat.ReviewAdditions,
+			ReviewDeletions: codeStat.ReviewDeletions,
+		}
+		cmsCodeStats = append(cmsCodeStats, cmsCodeStat)
+	}
+	return cmsCodeStats
+}
+
+func convertPRsToUserCodeStats(githubName string, month, year int, prs []codetracker.PullRequestInformation, reviews []codetracker.ReviewInformation) []user.CodeStats {
+	repoStats := make([]user.CodeStats, 0, 1048) // PNOOMA
+	for _, pr := range prs {
+		repoFound := false
+		for i, repoStat := range repoStats {
+			if repoStat.Repository == pr.Repository {
+				repoFound = true
+				repoStat.PRs = append(repoStat.PRs, pr.URL)
+				repoStat.MergedAdditions += pr.Additions
+				repoStat.MergedDeletions += pr.Deletions
+				repoStats[i] = repoStat
+				break
+			}
+		}
+		if !repoFound {
+			repoStat := user.CodeStats{
+				ID:              githubName + "-" + pr.Repository + "-" + strconv.Itoa(year) + "-" + strconv.Itoa(month),
+				GitHubName:      githubName,
+				Month:           month,
+				Year:            year,
+				PRs:             []string{pr.URL},
+				Repository:      pr.Repository,
+				MergedAdditions: pr.Additions,
+				MergedDeletions: pr.Deletions,
+			}
+			repoStats = append(repoStats, repoStat)
+		}
+	}
+	for _, review := range reviews {
+		repoFound := false
+		for i, repoStat := range repoStats {
+			if repoStat.Repository == review.Repository {
+				repoFound = true
+				repoStat.ReviewAdditions += int64(review.Additions)
+				repoStat.ReviewDeletions += int64(review.Deletions)
+				repoStat.Reviews = append(repoStat.Reviews, review.URL)
+				repoStats[i] = repoStat
+				break
+			}
+		}
+		if !repoFound {
+			repoStat := user.CodeStats{
+				ID:              githubName + "-" + review.Repository + "-" + strconv.Itoa(year) + "-" + strconv.Itoa(month),
```

Comment:
80 columns.  Bad readability.

```
			id := githubName + "-" + review.Repository + "-" +
				strconv.Itoa(year) + "-" + strconv.Itoa(month)
```
or

```
			id := fmt.Sprintf("%v-%v-%v-%v", githubName, review.Repository,
				strconv.Itoa(year), strconv.Itoa(month))
```

so that you can

```
			repoStat := user.CodeStats{
				ID:              id,
				GitHubName:      githubName,
```